Simulator/Core/GPUModel.cpp

```diff
@@ -53,28 +53,31 @@ void GPUModel::allocDeviceStruct()
 {
    // Allocate memory for random noise array
    int numVertices = Simulator::getInstance().getTotalVertices();
-   BGSIZE randNoise_d_size = numVertices * sizeof(float);   // size of random noise array
-   HANDLE_ERROR(cudaMalloc((void **)&randNoise_d, randNoise_d_size));
+   // BGSIZE randNoise_d_size = numVertices * sizeof(float);   // size of random noise array
+   // HANDLE_ERROR(cudaMalloc((void **)&randNoise_d, randNoise_d_size));
 
    // Allocate synapse inverse map in device memory
    allocEdgeIndexMap(numVertices);
+
+   // Create the CUDA stream used to launch synchronous GPU kernels during the simulation.
+   // This stream is passed to components like AllEdges and used consistently for kernel launches.
+   // For stream behavior and management, see:
+   // https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__STREAM.html
+   HANDLE_ERROR(cudaStreamCreate(&simulationStream_));
 }
 
 /// Copies device memories to host memories and deallocates them.
 void GPUModel::deleteDeviceStruct()
 {
    // Deallocate device memory
-   EdgeIndexMapDevice synapseIMapDevice;
-   HANDLE_ERROR(cudaMemcpy(&synapseIMapDevice, edgeIndexMapDevice_, sizeof(EdgeIndexMapDevice),
-                           cudaMemcpyDeviceToHost));
-   HANDLE_ERROR(cudaFree(synapseIMapDevice.outgoingEdgeBegin_));
-   HANDLE_ERROR(cudaFree(synapseIMapDevice.outgoingEdgeCount_));
-   HANDLE_ERROR(cudaFree(synapseIMapDevice.outgoingEdgeIndexMap_));
-   HANDLE_ERROR(cudaFree(synapseIMapDevice.incomingEdgeBegin_));
-   HANDLE_ERROR(cudaFree(synapseIMapDevice.incomingEdgeCount_));
-   HANDLE_ERROR(cudaFree(synapseIMapDevice.incomingEdgeIndexMap_));
-   HANDLE_ERROR(cudaFree(edgeIndexMapDevice_));
-   HANDLE_ERROR(cudaFree(randNoise_d));
+   vertices.deleteVerticesDeviceStruct(*allVerticesDevice);
+   // Copy device edge and vertex structs to host memory
+   edges.copyEdgeDeviceToHost(*allEdgesDevice);
+   // Deallocate device memory
+   edges.deleteEdgeDeviceStruct(*allEdgesDevice);
+   // HANDLE_ERROR(cudaFree(randNoise_d));
+   // closeFileMT();
+   HANDLE_ERROR(cudaStreamDestroy(simulationStream_));
```

Comment:
This method needs review. Questions include:
1. Where are `vertices` and `edges` declared?
2. Given the new `OperationManager` control that has all classes managing their own device memory, are any of these "non-local" deletions needed?